app/uk/gov/hmrc/exports/scheduler/jobs/PurgeAncientSubmissionsJob.scala

```diff
@@ -0,0 +1,132 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.exports.scheduler.jobs
+
+import org.bson.Document
+import org.bson.json.{JsonMode, JsonWriterSettings}
+import org.mongodb.scala.model.Filters._
+import play.api.Logging
+import uk.gov.hmrc.mongo.play.json.Codecs
+import play.api.libs.json.{Format, Json}
+import uk.gov.hmrc.exports.config.AppConfig
+import uk.gov.hmrc.exports.models.declaration.ExportsDeclaration
+import uk.gov.hmrc.exports.models.declaration.notifications.{ParsedNotification, UnparsedNotification}
+import uk.gov.hmrc.exports.repositories._
+import uk.gov.hmrc.exports.models.declaration.submissions.{Submission, SubmissionRequest}
+import uk.gov.hmrc.exports.mongo.ExportsClient
+import uk.gov.hmrc.mongo.workitem.WorkItem
+import uk.gov.hmrc.mongo.play.json.formats.MongoFormats
+
+import java.time._
+import javax.inject.{Inject, Singleton}
+import scala.collection.convert.DecorateAsScala
+import scala.concurrent.duration._
+import scala.concurrent.{ExecutionContext, Future}
+
+@Singleton
+class PurgeAncientSubmissionsJob @Inject() (
+  val appConfig: AppConfig,
+  submissionRepository: SubmissionRepository,
+  declarationRepository: DeclarationRepository,
+  unparsedNotificationRepository: UnparsedNotificationWorkItemRepository,
+  parsedNotificationRepository: ParsedNotificationRepository,
+  transactionalOps: PurgeSubmissionsTransactionalOps
+)(implicit ec: ExecutionContext)
+    extends ScheduledJob with ExportsClient with Logging with DecorateAsScala {
+
+  val submissionCollection = db.getCollection(submissionRepository.collectionName)
+  val declarationCollection = db.getCollection(declarationRepository.collectionName)
+  val notificationCollection = db.getCollection(parsedNotificationRepository.collectionName)
+  val unparsedNotificationCollection = db.getCollection(unparsedNotificationRepository.collectionName)
+
+  override val name: String = "PurgeAncientSubmissions"
+
+  override def interval: FiniteDuration = jobConfig.interval
+  override def firstRunTime: Option[LocalTime] = Some(jobConfig.elapseTime)
+
+  private def jsonSettings(mode: JsonMode) = JsonWriterSettings.builder.outputMode(mode).build
+
+  private val jobConfig = appConfig.purgeAncientSubmissions
+  private val clock: Clock = appConfig.clock
+
+  private val latestStatus = "latestEnhancedStatus"
+  private val statusLastUpdated = "enhancedStatusLastUpdated"
+
+  private val expiryDate = Codecs.toBson(ZonedDateTime.now(clock).minusDays(180))
+
+  private val latestStatusLookup =
+    or(equal(latestStatus, "GOODS_HAVE_EXITED"), equal(latestStatus, "DECLARATION_HANDLED_EXTERNALLY"), equal(latestStatus, "CANCELLED"))
```

Comment:
Where Tim wrote the ticket he's said REJECTED is also one of the statuses. I know that's not actually a valid enhanced status, but might he possibly mean any dec that has last been updated with a DMSREJ notification?
Because that would then include decs that have ERRORS enhanced status.

Reply:
EXPIRED_NO_ARRIVAL is another enhancedStatus that can result from a DMSREJ too

Reply:
Probably just need to clarify with him, which exact enahcnedStatuses do we want to purge

Reply:
Good point. As you say REJECTED is mapped to multiple statuses wanted clarification so took it out but forgot to make a note.